src/network/networkd-dhcp4.c

```diff
@@ -187,12 +187,37 @@ static int link_set_dhcp_routes(Link *link) {
         /* Clear old entries in case the set was already allocated */
         set_clear(link->dhcp_routes);
 
-        table = link_get_dhcp_route_table(link);
-
         r = sd_dhcp_lease_get_address(link->dhcp_lease, &address);
         if (r < 0)
                 return log_link_warning_errno(link, r, "DHCP error: could not get address: %m");
 
+        /* Now add here the prefix routes */
+        table = link_get_dhcp_route_table(link);
+        if (table != RT_TABLE_MAIN) {
```

Comment:
`table` may be set through VRF. In that case, `networkd` should not create the prefix route, and kernel creates it.